### ISSUE TITLE
Fix Request: false positive in BufferFilesNewStream

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#456]
IOUtils buffers the InputStream, so no extra buffering needed. OutputStream is however not buffered.
```
void ioUtilsBad(InputStream in, String path) throws IOException {
        OutputStream out = Files.newOutputStream(Paths.get(path)); // bad, not buffered
        IOUtils.copy(in, out);
    }
    void ioUtilsGood1(InputStream in, String path) throws IOException {
        OutputStream out = IOUtils.buffer(Files.newOutputStream(Paths.get(path)));
        IOUtils.copy(in, out);
    }
    void ioUtilsGood2(InputStream in, String path) throws IOException {
        OutputStream out = Files.newOutputStream(Paths.get(path));
        IOUtils.copy(in, IOUtils.buffer(out));
    }
    void ioUtilsGood3(String pathIn, String pathOut) throws IOException {
        InputStream in = Files.newInputStream(Paths.get(pathIn));
        OutputStream out = Files.newOutputStream(Paths.get(pathOut));
        IOUtils.copy(in, IOUtils.buffer(out)); // buffer inputstream
    }

```

**Comment by jborgers** [comment]
Tests show that buffering the output stream does not improve speed. The buffer can be written at once to the output stream, and not one by one. So, no need to buffer the outputstream in the copy.

**Comment by jborgers** [comment]
So all the below test cases are good, just the extra buffering is not needed and takes some more memory.
```
void ioUtilsGood(InputStream in, String path) throws IOException {
        OutputStream out = Files.newOutputStream(Paths.get(path)); // good, buffered
        IOUtils.copy(in, out);
    }
    void ioUtilsGood1(InputStream in, String path) throws IOException {
        OutputStream out = IOUtils.buffer(Files.newOutputStream(Paths.get(path)));
        IOUtils.copy(in, out);
    }
    void ioUtilsGood2(InputStream in, String path) throws IOException {
        OutputStream out = Files.newOutputStream(Paths.get(path));
        IOUtils.copy(in, IOUtils.buffer(out));
    }
    void ioUtilsGood3(String pathIn, String pathOut) throws IOException {
        InputStream in = Files.newInputStream(Paths.get(pathIn));
        OutputStream out = Files.newOutputStream(Paths.get(pathOut));
        IOUtils.copy(in, IOUtils.buffer(out)); // buffer inputstream
    }
}
```
